providers/implementations/ciphers/ciphercommon_block.c

```diff
@@ -105,33 +106,32 @@ void ossl_cipher_padblock(unsigned char *buf, size_t *buflen, size_t blocksize)
         buf[i] = pad;
 }
 
+/*
+ * ossl_cipher_unpadblock removes the padding from the final block in constant time.
+ * Note that if the padding is invalid, ERR_raise is not called in order to preserve
+ * constant-timedness.
+ */
 int ossl_cipher_unpadblock(unsigned char *buf, size_t *buflen, size_t blocksize)
 {
     size_t pad, i;
     size_t len = *buflen;
+    int gp = 1;
 
-    if(len != blocksize) {
-        ERR_raise(ERR_LIB_PROV, ERR_R_INTERNAL_ERROR);
-        return 0;
-    }
+    gp &= constant_time_eq_s(len, blocksize);
```

Comment:
Removing the return seems potentially problematic to me. This is clearly a misuse of the API, and should not result in a difference in timing.

Reply:
Fixed in commit `526350c`.